kamon-core/src/main/scala/kamon/trace/TracerModule.scala

```diff
@@ -122,12 +122,12 @@ private[kamon] class TracerModuleImpl(metricsExtension: MetricsModule, config: C
 
     val traceToken = token.getOrElse(newToken)
 
-    if (_settings.levelOfDetail == LevelOfDetail.MetricsOnly || !isLocal)
-      newMetricsOnlyContext(traceToken)
-    else {
-      if (!_settings.sampler.shouldTrace)
+    _settings.levelOfDetail match {
+      case LevelOfDetail.MetricsOnly ⇒
         newMetricsOnlyContext(traceToken)
-      else
+      case _ if !isLocal || !_settings.sampler.shouldTrace ⇒
+        newMetricsOnlyContext(traceToken)
+      case _ ⇒
         new TracingContext(traceName, traceToken, true, _settings.levelOfDetail, isLocal, startTimestamp, null, dispatchTracingContext)
```

Comment:
That null being passed there is surely a leftover from some refactored I did, and I apologize for it! But now that you are there, maybe you can help clean it out anyway :D.

In the previous PR you included a null check for the log instance which I don't see in this one. Maybe we should simply provide the TracerModule with a logging adapter via constructor param, WDYT?